packages/flutter/test/material/material_button_test.dart

```diff
@@ -67,4 +274,403 @@ void main() {
+            expectedClipPath,
+            areaToCompare: expectedClipRect.inflate(10.0),
+        ))
+        ..circle(color: directSplashColor)
+        ..rect(color: directHighlightColor),
+    );
+
+    const Color themeSplashColor1 = Color(0xFF001100);
+    const Color themeHighlightColor1 = Color(0xFF001100);
+
+    buttonWidget = Material(
+      child: Center(
+        child: MaterialButton(
+          onPressed: () { /* to make sure the button is enabled */ },
+          clipBehavior: Clip.antiAlias,
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Theme(
+          data: ThemeData(
+            highlightColor: themeHighlightColor1,
+            splashColor: themeSplashColor1,
+            materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          ),
+          child: buttonWidget,
+        ),
+      ),
+    );
+
+    expect(
+      Material.of(tester.element(find.byType(MaterialButton))),
+      paints
+        ..clipPath(pathMatcher: coversSameAreaAs(
+            expectedClipPath,
+            areaToCompare: expectedClipRect.inflate(10.0),
+        ))
+        ..circle(color: themeSplashColor1)
+        ..rect(color: themeHighlightColor1),
+    );
+
+    const Color themeSplashColor2 = Color(0xFF002200);
+    const Color themeHighlightColor2 = Color(0xFF002200);
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Theme(
+          data: ThemeData(
+            highlightColor: themeHighlightColor2,
+            splashColor: themeSplashColor2,
+            materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          ),
+          child: buttonWidget, // same widget, so does not get updated because of us
+        ),
+      ),
+    );
+
+    expect(
+      Material.of(tester.element(find.byType(MaterialButton))),
+      paints
+        ..circle(color: themeSplashColor2)
+        ..rect(color: themeHighlightColor2),
+    );
+
+    await gesture.up();
+  });
+
+  testWidgets('MaterialButton has no clip by default', (WidgetTester tester) async {
+    final GlobalKey buttonKey = GlobalKey();
+    final Widget buttonWidget = Material(
+      child: Center(
+        child: MaterialButton(
+          key: buttonKey,
+          onPressed: () { /* to make sure the button is enabled */ },
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Theme(
+          data: ThemeData(
+            materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          ),
+          child: buttonWidget,
+        ),
+      ),
+    );
+
+    expect(
+        tester.renderObject(find.byKey(buttonKey)),
+        paintsExactlyCountTimes(#clipPath, 0),
+    );
+  });
+
+  testWidgets('Disabled MaterialButton has correct default text color', (WidgetTester tester) async {
```

Comment:
This test seems repeated from `MaterialButton defaults` in the following lines: 
```
    // Disabled MaterialButton
    await tester.pumpWidget(
      const Directionality(
        textDirection: TextDirection.ltr,
        child: MaterialButton(
          onPressed: null,
          child: Text('button'),
        ),
      ),
    );
    /* ... */
    expect(material.textStyle.color, const Color(0x61000000));
```

It should be safe to delete this test